File: scaf-service/src/main/java/pe/utp/service/AsignacionService.java
package pe.utp.service;

import org.springframework.data.domain.Pageable;
import pe.utp.dto.PaginateResponseDto;
import pe.utp.dto.asignacion.AsignacionRequestDto;
import pe.utp.dto.asignacion.AsignacionResponseDto;

public interface AsignacionService {

    PaginateResponseDto<AsignacionResponseDto> listar(Pageable pageable);

    AsignacionResponseDto buscarPorId(Long id);

    AsignacionResponseDto crear(AsignacionRequestDto asignacionRequestDto);

    AsignacionResponseDto actualizar(Long id, AsignacionRequestDto asignacionRequestDto);

    void eliminar(Long id);
}

File: scaf-client/src/app/modules/asignacion/service/asignacion.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { AsignacionRequest, AsignacionResponse } from '../../../core/models/asignacion.model';
import { PaginateResponse } from '../../../core/models/paginate-response.model';

@Injectable({
  providedIn: 'root'
})
export class AsignacionService {
  private readonly apiUrl = 'http://localhost:9091/api/asignaciones';

  constructor(private readonly http: HttpClient) {}

  listar(page: number, size: number): Observable<PaginateResponse<AsignacionResponse>> {
    return this.http.get<PaginateResponse<AsignacionResponse>>(this.apiUrl, {
      params: { page, size }
    });
  }

  buscarPorId(id: number): Observable<AsignacionResponse> {
    return this.http.get<AsignacionResponse>(`${this.apiUrl}/${id}`);
  }

  crear(asignacion: AsignacionRequest): Observable<AsignacionResponse> {
    return this.http.post<AsignacionResponse>(this.apiUrl, asignacion);
  }

  actualizar(id: number, asignacion: AsignacionRequest): Observable<AsignacionResponse> {
    return this.http.put<AsignacionResponse>(`${this.apiUrl}/${id}`, asignacion);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
